### PR TITLE
Generic DAO error handling (GSI-1005)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Re-raises non-timeout PyMongo errors as DaoError.

### CODE REVIEW

File: src/hexkit/providers/mongodb/provider.py [hunk 1]
```diff
@@ -53,20 +54,20 @@
...
-    "translate_timeout_error",
+    "translate_pymongo_errors",
     "MongoDbDao",
 ]
 
 
 @contextmanager
-def translate_timeout_error():
+def translate_pymongo_errors():
     """Catch PyMongoError and re-raise it as DbTimeoutError if it is a timeout error."""
```
Comment (Cito) [inline src/hexkit/providers/mongodb/provider.py:64]:
```suggestion
    """Catch PyMongoError and re-raise it as appropriate DaoError subclass."""
```

File: tests/unit/test_dao.py [hunk 2]
```diff
@@ -157,20 +158,16 @@ async def test_db_timeout_error_translator():
     not_timeout_error = PyMongoError()  # .timeout is False by default
 
     # Non-timeout errors should be re-raised as they are
```
Comment (Cito) [inline tests/unit/test_dao.py:160]:
```suggestion
    # Non-timeout errors should be translated to base DaoErrors
```

**Review by Cito — CHANGES_REQUESTED** [review]
Looks good, but docstrings/comments still need to be adapted.

File: tests/unit/test_dao.py (hunk 2 above)
Comment (TheByronHimes) [reply tests/unit/test_dao.py]: See [5daea3c](https://github.com/ghga-de/hexkit/pull/133/commits/5daea3c63eba2fda08ed7925a13e10ffa9883550) for a fix

File: src/hexkit/providers/mongodb/provider.py (hunk 1 above)
Comment (TheByronHimes) [reply src/hexkit/providers/mongodb/provider.py]: Fixed in [5daea3c](https://github.com/ghga-de/hexkit/pull/133/commits/5daea3c63eba2fda08ed7925a13e10ffa9883550)

**Review by Cito — APPROVED** [review]

**Pull request merged**